Wikipedia/Code/WMFSettingsViewController.m

```diff
@@ -1,634 +1,296 @@
         @strongify(self)
         [self dismissViewControllerAnimated : YES completion : nil];
     }];
     self.navigationItem.leftBarButtonItems = @[xButton];
-
-    self.highlightedTextAttributes = @{ NSFontAttributeName: [UIFont italicSystemFontOfSize:MENU_TITLE_FONT_SIZE] };
-
-    self.view.backgroundColor = [UIColor wmf_settingsBackgroundColor];
-
-    self.scrollView.clipsToBounds    = NO;
-    self.scrollView.minSubviewHeight = 45;
-
-    self.rowViews = @[].mutableCopy;
 }
 
-- (void)viewWillAppear:(BOOL)animated {
-    [super viewWillAppear:animated];
-
-    [self.rowViews removeAllObjects];
-
-    [self loadRowViews];
-
-    self.scrollView.orientation     = TABULAR_SCROLLVIEW_LAYOUT_HORIZONTAL;
-    self.scrollView.tabularSubviews = self.rowViews;
-}
-
-- (void)viewDidAppear:(BOOL)animated {
-    [super viewDidAppear:animated];
-
-    [[NSNotificationCenter defaultCenter] addObserver:self
-                                             selector:@selector(tabularScrollViewItemTappedNotification:)
-                                                 name:@"TabularScrollViewItemTapped"
-                                               object:nil];
-}
-
-- (void)viewWillDisappear:(BOOL)animated {
-    [[NSNotificationCenter defaultCenter] removeObserver:self
-                                                    name:@"TabularScrollViewItemTapped"
-                                                  object:nil];
-    [super viewWillDisappear:animated];
+- (NSString*)title {
+    return MWLocalizedString(@"settings-title", nil);
 }
 
-#pragma mark - Access
-
-- (SecondaryMenuRowIndex)getIndexOfRow:(NSDictionary*)row {
-    return (SecondaryMenuRowIndex)((NSNumber*)row[@"tag"]).integerValue;
+-(void)configureTableDataSource {
+    self.elementDataSource                  = [[SSSectionedDataSource alloc] init];
+    self.elementDataSource.rowAnimation     = UITableViewRowAnimationNone;
+    self.elementDataSource.tableView        = self.tableView;
+    self.elementDataSource.cellClass        = [WMFSettingsTableViewCell class];
+    self.elementDataSource.tableActionBlock = ^BOOL (SSCellActionType action, UITableView* tableView, NSIndexPath* indexPath) {
+        return NO;
+    };
+    
+    @weakify(self)
+    self.elementDataSource.cellConfigureBlock = ^(WMFSettingsTableViewCell* cell, WMFSettingsMenuItem* menuItem, UITableView* tableView, NSIndexPath* indexPath) {
+        
+        cell.title          = menuItem.title;
+        cell.iconColor      = menuItem.iconColor;
+        cell.iconName       = menuItem.iconName;
+        cell.disclosureType = menuItem.disclosureType;
+        cell.disclosureText = menuItem.disclosureText;
+        
+        cell.selectionStyle = (menuItem.disclosureType == WMFSettingsMenuItemDisclosureType_Switch) ? UITableViewCellSelectionStyleNone : UITableViewCellSelectionStyleDefault;
+        
+        @strongify(self)
+        [cell.disclosureSwitch setOn:[self getSwitchOnValueForMenuItemType:menuItem.type]];
```

Comment:
Its better if this piece of information should be inside the WMFSettingsItem that way each cell is driven solely by the values stored in the model object.